src/main/java/hudson/plugins/copyartifact/CopyArtifact.java

```diff
@@ -132,7 +132,7 @@ public CopyArtifact(String projectName) {
         // check the permissions only if we can
         StaplerRequest req = Stapler.getCurrentRequest();
         if (req!=null) {
-            AbstractProject<?,?> p = req.findAncestorObject(AbstractProject.class);
+            Job p = req.findAncestorObject(Job.class);
```

Comment:
This (and only this) change isn't applied in 1.35.1 as @jglick points.
[JENKINS-28247](https://issues.jenkins-ci.org/browse/JENKINS-28247) may be caused for that.